src/main/java/net/mcreator/plugin/PluginLoader.java

```diff
@@ -184,4 +188,26 @@ public PluginLoader() {
 		return plugin;
 	}
 
+	public void checkPluginUpdates() {
+		for (Plugin plugin : plugins) {
+			if (MCreatorApplication.isInternet) {
+				if (plugin.getInfo().getUpdateUrl() != null) {
+					if (!plugin.getInfo().getVersion().equals("not specified")) {
+						JsonObject version = new Gson()

```

Comment:
I meant to use GSON here to parse update JSON directly to eg. PluginUpdateInfo object, not that you do it, but use GSON for this so we don't reinvent the wheel.

Reply:
I really don't know how to do it, so I tried something else.